lib/concurrent/mvar.rb

```diff
@@ -68,10 +68,14 @@ def take(timeout = nil)
 
         # If we timed out we'll still be empty
         if unlocked_full?
-          value = @value
-          @value = EMPTY
-          @empty_condition.signal
-          apply_deref_options(value)
+          if block_given?
+            yield @value
```

Comment:
Sorry for the space styling. I'll change this.

I'm not sure if I understand correctly your question. 

I've tried to reproduce the behavior of IO#open for instance:
 * When no block is given, it returns a file.
 * When a block is given, the file is given to this block. 

In the MVar#take case : 
 * when no block is given: it returns the value and remove it from the MVar (I haven't changed this behavior).
 * When a block is given, the value is given to the block but not removed from the MVar.

I'm wondering if apply_deref_options should be called when passing the value to the block. I've chosen not to call it, like it is done in MVar#modify.